packer/box_handler.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash +e
+
+# To resolve error message as below, which is a dirty worksspace
+# Build 'virtualbox-ovf' errored: Error enabling VRDP: VBoxManage error: VBoxManage: error: The machine 'rackhd-ubuntu-14.04' is already locked for a session (or being unlocked)
+virtualBoxDestroyAll() {
+    set +e
+    for uuid in `vboxmanage list vms | awk '{print $2}' | tr -d '{}'`; do
+        echo "shutting down vm ${uuid}"
+        vboxmanage controlvm ${uuid} poweroff
+        echo "deleting vm ${uuid}"
+        vboxmanage unregistervm ${uuid}
+    done
+    pkill packer
+    set -e
+}
+
+# To resolve issue during packer build:
+# Machine settings file '/home/jenkins/VirtualBox VMs/rackhd-ubuntu-14.04/rackhd-ubuntu-14.04.vbox' already exists
+removeOldBox()
+{
+    local OS_VER=ubuntu-14.04
+    rm -rf  ~/VirtualBox\ VMs/rackhd-${OS_VER}
+    OS_VER=ubuntu-16.04
+    rm -rf  ~/VirtualBox\ VMs/rackhd-${OS_VER}
+}
+
+#########################################
+#
+# Clean up box
+#
+#########################################
+cleanUpBox(){
+    virtualBoxDestroyAll
+    removeOldBox
+}
+
+#########################################
+#
+#  check dependencies
+#
+#########################################
+checkDependencies(){
+    set +e
+    echo "[Info] Start to check the dependencies"
+    packer_path=$( which packer )
+    if [ -z "$packer_path" ];then
+        echo "[ERROR] packer is not installed . please install it."
+        exit 5
+    fi
+    vagrant_path=$( which vagrant )
+    if [ -z "$vagrant_path" ];then
+        echo "[ERROR] vagrant is not installed . please install it."
+        exit 5
+    fi
+    jq_path=$(which jq)
+    if [ ! -n "$(which jq)" ] ; then
```

Comment:
are you going to use   ```if [ -z $jq_path ] ; then``` like what you did before?

Reply:
Thanks, fixed.